Don't build non-existent image URLs

chickdan opened the pull request:
I believe this fixes #884 as an image URL will only be returned if there are image tags, and without a URL LazyImage should result in the failure case. I don't have a Jellyfin server setup so I could be wrong with my changes so guidance would be appreciated.

LePips commented on Shared/Extensions/JellyfinAPI/BaseItemDto+Images.swift:
```diff
@@ -80,11 +80,13 @@ extension BaseItemDto {
         let scaleWidth = maxWidth == nil ? nil : UIScreen.main.scale(maxWidth!)
         let scaleHeight = maxHeight == nil ? nil : UIScreen.main.scale(maxHeight!)
-        let tag = imageTags?[type.rawValue]
+        let tag = imageTags[type.rawValue]
```
Checking the `imageTags` is the first step, the next is checking if this tag has a value and if not also `return nil`

LePips replied:
Additionally, we will have to make sure the fields for the tags are requested along with the items.

LePips commented:
So currently, that fix would actually break backdrop images. I took a look again at these fields because I forgot they're split, `imageTags` and `backdropImageTags` to say a few. Backdrop image tags aren't in `imageTags`, so this will always fail for them. We will have to switch on the type of image being requested and retrieve + check in the right place. In the case for backdrops, we will always take the first (primary) image. This has always worked before because calling for an item backdrop image without an index/tag will just return the primary image.

chickdan commented:
> We will have to switch on the type of image being requested and retrieve + check in the right place.

Added a new function that switches on `ImageType`, most of the types do not have corresponding <>ImageTag fields so I default to `imageTags` as that's also how the behavior was previously.

> Backdrop image tags aren't in `imageTags`, so this will always fail for them... In the case for backdrops, we will always take the first (primary) image.

For this, I return `backdropImageTags.first` from the new `getImageTag()` function.

> This has always worked before because calling for an item backdrop image without an index/tag will just return the primary image.

Considering the existing function always worked with a `nil` tag as long as the type was `.backdrop` I added a check to allow the request to be made.

Hopefully, I understood the requirements 🙂 let me know if further adjustments are required.

LePips commented on Shared/Extensions/JellyfinAPI/BaseItemDto+Images.swift:
```diff
@@ -80,11 +80,16 @@ extension BaseItemDto {
-        let tag = imageTags?[type.rawValue]
+        let tag = getImageTag(for: type)
+
+        if tag == nil && type != .backdrop {
```
Can remove this and just have:
```swift
guard let tag else { return nil }
```

We don't need the check for backdrop images, if there aren't any then they won't populate the image tags. I only mentioned the "lack of index returning the primary image" as a lucky coincidence with the API.
